datafoundation/templates_storagecluster.go

```diff
@@ -0,0 +1,168 @@
+	MaxSkew:           1,
+	TopologyKey:       "topology.kubernetes.io/zone",
+	WhenUnsatisfiable: corev1.DoNotSchedule,
+	LabelSelector: &metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				// Key is from rook/pkg/operator/ceph/cluster/osd/labels.go
+				Key:      "ceph.rook.io/pvc",
+				Operator: metav1.LabelSelectorOpExists,
+			},
+		},
+	},
+}
+
+var preparePlacementTSC corev1.TopologySpreadConstraint = corev1.TopologySpreadConstraint{
+	MaxSkew:           1,
+	TopologyKey:       "kubernetes.io/hostname",
+	WhenUnsatisfiable: corev1.DoNotSchedule,
+	LabelSelector: &metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				// Key is from rook/pkg/operator/ceph/cluster/osd/labels.go
+				Key:      "ceph.rook.io/pvc",
+				Operator: metav1.LabelSelectorOpExists,
+			},
+		},
+	},
+}
+
+var StorageClusterTemplate = ocsv1.StorageCluster{
+	Spec: ocsv1.StorageClusterSpec{
+		// The label selector is used to select only the worker nodes for
+		// both labeling and scheduling.
+		LabelSelector: &metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{{
+				Key:      "node-role.kubernetes.io/worker",
+				Operator: metav1.LabelSelectorOpExists,
+			}, {
+				Key:      "node-role.kubernetes.io/infra",
+				Operator: metav1.LabelSelectorOpDoesNotExist,
+			}},
+		},
+		ManageNodes: false,
+		MonPVCTemplate: &corev1.PersistentVolumeClaim{
+			Spec: corev1.PersistentVolumeClaimSpec{
+				StorageClassName: &gp2,
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						"storage": resource.MustParse("50Gi"),
+					},
+				},
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					corev1.ReadWriteOnce,
+				},
+			},
+		},
+		Resources: map[string]corev1.ResourceRequirements{
+			"mds":            GetResourceRequirements("mds"),
+			"mgr":            GetResourceRequirements("mgr"),
+			"mon":            GetResourceRequirements("mon"),
+			"crashcollector": GetResourceRequirements("crashcollector"),
+		},
+		StorageDeviceSets: []ocsv1.StorageDeviceSet{{
+			Name:  "default",
+			Count: 1,
+			DataPVCTemplate: corev1.PersistentVolumeClaim{
+				Spec: corev1.PersistentVolumeClaimSpec{
+					StorageClassName: &gp2,
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					},
+					VolumeMode: &volumeModeBlock,
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							"storage": resource.MustParse(fmt.Sprintf("%dTi", OSDSizeInTiB)),
+						},
+					},
+				},
+			},
+			Placement: rook.Placement{
+				TopologySpreadConstraints: []corev1.TopologySpreadConstraint{
+					commonTSC,
+				},
+			},
+			PreparePlacement: rook.Placement{
+				TopologySpreadConstraints: []corev1.TopologySpreadConstraint{
+					commonTSC,
+					preparePlacementTSC,
+				},
+			},
+			Portable:    true,
+			Replica:     3,
+			DeviceClass: "ssd",
```

Comment:
Why is this called ssd? was this the same way with the deployer?
Also, can we move this and `replica` after `count`?

Reply:
We didn't have DeviceClass in the deployer. Migration team gave me this config.
Yes, I can move the fields but cannot rename the DeviceClass